pkg/operators/combiner/combiner.go

```diff
@@ -0,0 +1,309 @@
+			combinedDs.Release(combinedPacket)
+		}
+	}()
+
+	targetCount := 0
+
+	var c <-chan time.Time
+
+	if interval == 0 {
+		// Define a maximum waiting time for data from all targets
+		// TODO: Make it configurable?
+		timeout := time.NewTimer(5 * time.Second)
+		defer timeout.Stop()
+		c = timeout.C
+	} else {
+		// Even if we receive data from all targets, we emit the combined data
+		// only after the requested interval
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+		c = ticker.C
+	}
+
+	emitAndAllocate := func() error {
+		gadgetCtx.Logger().Debug("combiner: emitting data array")
+
+		if err := combinedDs.EmitAndRelease(combinedPacket); err != nil {
+			gadgetCtx.Logger().Errorf("failed to emit data array for ds combiner %s: %s",
+				combinedDs.Name(), err)
+		}
+
+		// Allocate a new packet array for next iteration
+		combinedPacket, err = combinedDs.NewPacketArray()
+		if err != nil {
+			// We can't continue without a new packet array
+			return fmt.Errorf("creating new packet array: %w", err)
+		}
+
+		targetCount = 0
+		return nil
+	}
+
+	for {
+		select {
+		case <-o.done:
+			gadgetCtx.Logger().Debug("combiner: done")
+			return
+		case inPacket := <-packetBuf:
+			if inPacket == nil {
+				continue
+			}
+
+			targetCount++
+			gadgetCtx.Logger().Debugf("combiner: received data from %d/%d targets", targetCount, o.targets)
+
+			// TODO: Support Append multiple packets at once?
+			for i := 0; i < inPacket.Len(); i++ {
+				combinedPacket.Append(inPacket.Get(i))
+			}
+
+			// For data sources that doesn't have an interval, we wait for data
+			// from all targets before emitting the combined data.
+			if interval == 0 && targetCount == o.targets {
+				if err := emitAndAllocate(); err != nil {
+					gadgetCtx.Logger().Errorf("emitting and allocating data: %s", err)
+				}
+				return
+			}
+		case <-c:
+			if interval == 0 {
+				gadgetCtx.Logger().Warnf("Data is incomplete: timeout waiting for data from all targets (%d/%d)",
+					targetCount, o.targets)
+			}
+
+			if err := emitAndAllocate(); err != nil {
+				gadgetCtx.Logger().Errorf("emitting and allocating data: %s", err)
+				return
+			}
+		}
+	}
+}
+
+func (o *combinerOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error {
+	o.done = make(chan struct{})
+
+	for _, ds := range gadgetCtx.GetDataSources() {
+		if ds.Type() != datasource.TypeArray {
+			continue
+		}
+
+		interval, err := getFetchAnnotation(ds)
+		if err != nil {
+			return fmt.Errorf("getting fetch annotation for ds %s: %w", ds.Name(), err)
+		}
+
+		// Disable original data source to avoid other operators subscribing to it
+		ds.SetRequested(false)
+
+		// Register a new data source that will emit the combined data
+		combinedDs, err := gadgetCtx.RegisterDataSource(
+			datasource.TypeArray,
+			fmt.Sprintf("%s-%s", DataSourcePrefix, ds.Name()),
+		)
+		if err != nil {
+			return fmt.Errorf("registering combiner data source for %s: %w", combinedDs.Name(), err)
```

Comment:
```suggestion
			return fmt.Errorf("registering combiner data source for %s: %w", ds.Name(), err)
```
Otherwise this would probably crash on err != nil.